packages/common/src/services/audius-backend/solana.ts

```diff
@@ -757,3 +713,210 @@ export const createVersionedTransaction = async (
     addressLookupTableAccounts
   }
 }
+
+// NOTE: The above all need to be updated to use SDK. The below is fresh.
+
+/**
+ * In the case of a failed Coinflow withdrawal, transfers the USDC back out of
+ * the root Solana account and into the user's user bank account.
+ *
+ * Note that this uses payment router to do the transfer, so that indexing sees
+ * this transfer and handles it appropriately.
+ */
+export const recoverUsdcFromRootWallet = async ({
+  sdk,
+  sender,
+  receiverEthWallet,
+  amount
+}: {
+  sdk: AudiusSdk
+  sender: Keypair
+  receiverEthWallet: string
+  amount: bigint
+}) => {
+  const { userBank } =
+    await sdk.services.claimableTokensClient.getOrCreateUserBank({
+      ethWallet: receiverEthWallet,
+      mint: 'USDC'
+    })
+
+  // See: https://github.com/solana-labs/solana-program-library/blob/d6297495ea4dcc1bd48f3efdd6e3bbdaef25a495/memo/js/src/index.ts#L27
+  const memoInstruction = new TransactionInstruction({
+    keys: [
+      {
+        pubkey: sender.publicKey,
+        isSigner: true,
+        isWritable: true
+      }
+    ],
+    programId: MEMO_PROGRAM_ID,
+    data: Buffer.from(RECOVERY_MEMO_STRING)
+  })
+  const transferInstruction =
+    await sdk.services.paymentRouterClient.createTransferInstruction({
+      total: amount,
+      sourceWallet: sender.publicKey,
+      mint: 'USDC'
+    })
+  const routeInstruction =
+    await sdk.services.paymentRouterClient.createRouteInstruction({
+      total: amount,
+      splits: [
+        {
+          amount,
+          wallet: userBank
+        }
+      ],
+      mint: 'USDC'
+    })
+  const transaction = await sdk.services.claimableTokensClient.buildTransaction(
+    { instructions: [memoInstruction, transferInstruction, routeInstruction] }
+  )
+  transaction.sign([sender])
+  const signature = await sdk.services.paymentRouterClient.sendTransaction(
+    transaction,
+    { skipPreflight: true }
+  )
+  return signature
+}
+
+/**
+ * Transfers tokens out of a user bank.
+ * Notes:
+ * - Including a signer will mark this transfer as a "withdrawal preparation"
+ *   by signing a memo indicating such. This prevents the transfer from showing
+ *   as a withdrawal on the withdrawal history page.
+ * - Users have restrictions on creating token accounts via relay, so if the
+ *   destination token account doesn't exist this might fail.
+ */
+export const transferFromUserBank = async ({
+  sdk,
+  mint,
+  connection,
+  amount,
+  ethWallet,
+  destinationWallet,
+  track,
+  make,
+  analyticsFields,
+  signer
+}: {
+  sdk: AudiusSdk
+  mint: PublicKey
+  connection: Connection
+  amount: number
+  ethWallet: string
+  destinationWallet: PublicKey
+  track: CommonStoreContext['analytics']['track']
```

Comment:
Do these things not have their own types outside of the context? Should probably use those directly unless the intention is really to hard-couple this to whatever happens to be in the StoreContext.

Reply:
No they don't, at least from what I could tell